File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int start = 0;
        int next = start + 1;
        while (start < next && next < nums.length) {
            if (nums[start] == nums[next]) {
                next++;
            } else {
                nums[++start] = nums[next];
                next++;
            }
        }
        return ++start;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int j = 1;
        for (int i = 0; i < nums.size()-1; i++) {
            if (nums[i] != nums[i + 1]) {
                nums[j] = nums[i + 1];
                j++;
            }
        }
        return j;
    }
};